general/mem_manager.hpp

```diff
@@ -112,11 +112,22 @@ MemoryClass operator*(MemoryClass mc1, MemoryClass mc2);
     In many respects this class behaves like a pointer:
     * When destroyed, a Memory object does NOT automatically delete any
       allocated memory.
-    * Only the method Delete() will deallocate a Memory object.
-    * Other methods that modify the object (e.g. New(), Wrap(), etc) will simply
-      overwrite the old contents.
-    * One difference with a pointer is that a const Memory object does not allow
-      modification of the content (unlike e.g. a const pointer).
+    * Only the method `Delete()` will deallocate a Memory object.
+    * Other methods that modify the object (e.g. `New()`, `Wrap()`, etc) will
+      simply overwrite the old contents.
+    In other aspects this class differs from a pointer:
+    * Pointer arithmetic is not supported, `MakeAlias()` should be used instead.
+    * Const Memory object does not allow modification of the content
+      (unlike e.g. a const pointer).
+    * Move constructor and assignement will transfer ownership flags, and
+      `Reset()` the moved Memory object.
+    * Copy constructor and assignement copy flags. This may result in two Memory
+      objects owning the data which is an invalid state. This invalid state MUST
+      be resolved by users manually using `SetHostPtrOwner()`,
+      `SetDevicePtrOwner()`, or `ClearOwnerFlags()`.
+    * When creating alias memory objects, the consistency of memory flags have
+      to be manually taken care of using either `Sync()` or `SyncAlias`. Failure
+      to do so will result in silent misuse of unsynchronized data.
```

Comment:
To be more precise: "When moving or copying (between host and device) alias Memory objects and/or their base Memory objects, ...."

Reply:
"SyncAlias" --> "SyncAlias()"

Reply:
done in [418054b](https://github.com/mfem/mfem/pull/2355/commits/418054b770a4cac104a33b46f18dce432a3f5482)